### PR TITLE
Add thread local variable to identify tenant admin intitializtaion op…

### PR DESCRIPTION
**Pull request by IsuraD**
…erations.

Fixing 	https://wso2.org/jira/browse/IDENTITY-4752

### CODE REVIEW

File: components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java [hunk 1]
```diff
@@ -77,8 +81,12 @@ public String addTenant(TenantInfoBean tenantInfoBean) throws Exception {
         PrivilegedCarbonContext carbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
         carbonContext.setTenantDomain(tenantDomain);
         carbonContext.setTenantId(tenantId);
+
         TenantMgtUtil.addClaimsToUserStoreManager(tenant);
 
+        //Remove thread local variable set to identify operation triggered for a tenant admin user.
+        TenantMgtUtil.clearTenantAdminCreationOperation();
+
         PrivilegedCarbonContext.endTenantFlow();
```
Comment (johannnallathamby) [inline components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java]: Can we end the tenant flow inside a finally block

File: components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java [hunk 2]
```diff
@@ -77,8 +81,12 @@ public String addTenant(TenantInfoBean tenantInfoBean) throws Exception {
         PrivilegedCarbonContext carbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
         carbonContext.setTenantDomain(tenantDomain);
         carbonContext.setTenantId(tenantId);
+
         TenantMgtUtil.addClaimsToUserStoreManager(tenant);
 
+        //Remove thread local variable set to identify operation triggered for a tenant admin user.
+        TenantMgtUtil.clearTenantAdminCreationOperation();
```
Comment (johannnallathamby) [inline components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java]: Why not clear the threadlocal in a finally block as the last line of the method

File: components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java [hunk 3]
```diff
@@ -65,21 +65,32 @@ public String addTenant(TenantInfoBean tenantInfoBean) throws Exception {
...
-        carbonContext.setTenantId(tenantId);
-        TenantMgtUtil.addClaimsToUserStoreManager(tenant);
+            PrivilegedCarbonContext carbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
+            carbonContext.setTenantDomain(tenantDomain);
+            carbonContext.setTenantId(tenantId);
+
+            TenantMgtUtil.addClaimsToUserStoreManager(tenant);
+        } finally {
+            //Remove thread local variable set to identify operation triggered for a tenant admin user.
+            TenantMgtUtil.clearTenantAdminCreationOperation();
```
Comment (johannnallathamby) [inline components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java]: Can we move this to the end of the method to another finally block

**Comment by callkalpa** [comment]
This PR has API changes, hence this should go into the master or the release-4.6.0 branch (which is the next minor release). Since this PR is sent to 4.5.x branch, I cherry-picked the two commits with 24917bbcdb33b4891268b1d4ec86b8f73cf856a5 and 62715b7fac7e1efbb49c2f5666e3e9c307dd9d8b

Since the fix is applied to the release-4.6.0 branch (and will be available in 4.6.0 release), closing this PR.

**Pull request closed without merging**